envsec/internal/auth/cache.go

```diff
@@ -0,0 +1,88 @@
+package auth
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/MicahParks/keyfunc/v2"
+	"github.com/pkg/errors"
+)
+
+const dirName = ".jetpack"
+const jwksFileName = "jwks.json"
+const cacheDuration = 1 * time.Hour
+
+func (a *Authenticator) fetchJWKSWithCache() (*keyfunc.JWKS, error) {
+	jwksURL := fmt.Sprintf("https://%s/.well-known/jwks.json", a.Domain)
+	wd, err := os.Getwd()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	path := filepath.Join(wd, dirName, jwksFileName)
```

Comment:
Use `os.UserCacheDir()` (similar to `getAuthFilePath`) that way multiple projects can share jwks.

The file should look like:

`filepath.Join(os.UserCacheDir(), "jwks", domain)` or something like that. That way all tools can share cache.

Reply:
I went with 
```go
// example ~/.cache/.jetpack/auth.jetpack.io.jwks.json
path := filepath.Join(cacheBaseDir, dirName, cacheFileName)
```
cacheFileName includes the domain. Both auth0 and stytch idp domains can share the same directory but different file names.